packages/redis/test/Adaptor.test.js

```diff
@@ -495,3 +496,67 @@ describe('jGet', () => {
     });
   });
 });
+
+describe('jSet', () => {
+  const jSetClient = {
+    json: {
+      set: async (key, path, values) => {
+        expect(path).to.eql('$');
+        expect(key).to.eql('animals:1');
+        expect(values).to.eql({ name: 'mammoth' });
+        return 'OK';
+      },
+    },
+  };
+  it('should expand references', async () => {
+    setMockClient(jSetClient);
+    const state = { value: { name: 'mammoth' } };
+    await jSet('animals:1', s => s.value)(state);
+  });
+
+  it('should throw if existing key has a different redis type', async () => {
+    setMockClient({
+      json: {
+        set: () => {
+          throw new Error('Existing key has wrong Redis type');
+        },
+      },
+    });
+
+    try {
+      await jSet('hash-key', { name: 'mammoth' })({});
+    } catch (error) {
+      expect(error.message).to.eql('Existing key has wrong Redis type');
+    }
+  });
+  it('should throw if invalid params', async () => {
+    setMockClient({
+      json: {
+        set: () => {
+          throw new Error();
+        },
+      },
+    });
+
+    const state = {};
+    try {
+      await jSet()(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+      expect(error.code).to.eql('ARGUMENT_ERROR');
+    }
+  });
+
+  it('should always sets results at the document root', async () => {
```

Comment:
@mtuchi I've updated this test to what I meant.

Notice that it's only asserting one thing. That's the only thing this test cares about. Super focused.

It's technically redundant because we've already got an implied test for it, but for me it's worth calling out like this

Reply:
Thanks @josephjclark that makes lot of sense